docs/devel/olm-integration.md

```diff
@@ -0,0 +1,151 @@
+# KubeVirt's OLM and Operator Marketplace Integration
+
+## Introduction
+
+### Operator Lifecycle Manager (OLM)
+
+https://github.com/operator-framework/operator-lifecycle-manager
+
+OLM is the Operator Lifecycle Manager, which consists of 2 operators:
+
+#### OLM Operator
+
+Installs application operators based in information in ClusterServiceVersions
+
+CRDs:
+
+- ClusterServiceVersion (CSV):  
+  contains application metadata: name, version, icon, required resources, installation, etc...
+  provided by developer, together with CRD declarations and package description. The latter declares channels and their CSV version.
+  installed by Catalog Operator
+
+- OperatorGroup:  
+  declares on which namespaces OLM should operate
+  provided and installed by developer, or in UI
+
+#### Catalog Operator
+
+Prepares installations of operators by installing the application's CRDs and CSVs
+
+CRDs:
+
+- CatalogSource:  
+  declares available packages
+  provided and installed by Marketplace Operator based on CatalogSourceConfig
+
+- Subscription:  
+  declares which version of an operator to install (which channel from which source)
+  provided and installed by developer, or in UI
+
+- InstallPlan:  
+  calculated list of resources to be created in order to automatically install/upgrade a CSV
+  created and insalled by the Catalog Operator, needs manual or automatic approval
+
+### Operator Marketplace
+
+https://github.com/operator-framework/operator-marketplace
+
+The Operator Marketplace has another operator
+
+CRDs:
+
+- OperatorSource:  
+  declares where to find applications bundles (CSV + CRD + package)
+  provided and installed by developer, and/or already installed pointing to official repositories (community operators)
+
+- CatalogSourceConfig:  
+  declares which packages to enable in the marketplace
+  created and deployed by marketplace operator
+
+## KubeVirt Manifests
+
+Our OLM / Marketplace manifest templates live in /manifests/release/olm. As for all manifests, you need to run
+`make generate && make manifests` for getting their final version in the `_out/` directory.
+
+The bundle subdirectory contains:
+  - the ClusterServiceVersion manifest
+  - the CRD manifest
+  - the Package manifest: this contains the available distribution channels and their corresponding CSV name
+  These files are pushed to Quay (after they are processed with)
+
+Then we have:
+  - the OperatorSource manifest: this will be deployed to your cluster.
```

Comment:
"deploy to" - is a remote operation
"deploy on" - a local one
which one is the case here?

Reply:
well, no clue... We don't know if it's a remote or local cluster.
But thanks for your review! I think I resolved all other comments.